### ISSUE TITLE
Stop raising an error when glacier exceeds domain boundaries?

### ISSUE THREAD

**Issue by fmaussion** [OGGM/oggm#330]
The current behavior is to raise an error when this happens:

```python
        if self.fls[-1].thick[-1] > 10:
            if not self.is_tidewater:
                raise RuntimeError('Glacier exceeds domain boundaries.')
```

Actually, the model can still run when ice is at the boundaries (at least with the default numerical scheme, which is stable is this case).

If we change this the model output will be badly wrong though - it will be the users responsibility to check that all the glaciers are valid...

@bmarzeion @anoukvlug thoughts?

**Comment by anoukvlug** [comment]
Why would you want to change this? So far I found it a very useful error message. I don't think I understand what the benefit could be of removing it.

**Comment by fmaussion** [comment]
The reasoning was that sometimes the glaciers grow out of boundaries for a few years and then the program stops for this glacier, although it could continue to run.

Anyways I think you are right, the default should remain to throw an error: I will add an option to switch it off though.

**Comment by anoukvlug** [comment]
I agree that it would be good to keep the error as a default. It is an error that is easy to solve by increasing the domain. Therefore I am not sure how much value the switch option will have, but it might be useful in some cases to have the choice.
